cms-2017-simulated-datasets/code/create_parent_dicts.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+import datetime
+import fnmatch
+import os
+import re
+import requests
+import subprocess
+import urllib3
+
+from dataset_records import *
+from mcm_store import get_mcm_dict
+from utils import get_from_deep_json
+from das_json_store import get_parent_dataset
+from eos_store import check_datasets_in_eos_dir
+
+def main(datasets):
+    nano_to_mini = dict()
+    mini_to_nano = dict()
+    
+    for dataset in datasets:
+        if dataset.endswith('NANOAODSIM'):
+            nano = dataset
+            mini = get_parent_dataset(nano, das_dir= './inputs/das-json-store')
+            try:
+                
+                if mini.endswith('/AODSIM'):
+                    result = subprocess.run(f'dasgoclient -query="parent dataset={nano}"', capture_output= True, shell= True, text= True).stdout.split('\n')
+                    for line in result:
+                        if line.endswith('MINIAODSIM'):
+                            mini = line
```

Comment:
Possibly print a warning that Nano with an AODSIM parent has been found so that we can track and check afterwards (print the input dataset name so that we see where it comes from)